--- a/paper-server/build.gradle.kts
+++ b/paper-server/build.gradle.kts
@@ -25,6 +_,16 @@
     updatingMinecraft {
         // oldPaperCommit = "7e80cef5198561d0db53406127e5b8bc7af51577"
     }
+
+    val kitty = forks.register("kitty") {
+        upstream.patchDir("paperServer") {
+            upstreamPath = "paper-server"
+            excludes = setOf("src/minecraft", "patches", "build.gradle.kts")
+            patchesDir = rootDirectory.dir("kitty-server/paper-patches")
+            outputDir = rootDirectory.dir("paper-server")
+        }
+    }
+    activeFork = kitty
 }
 
 tasks.generateDevelopmentBundle {
@@ -86,7 +_,21 @@
     }
 }
 
-val log4jPlugins = sourceSets.create("log4jPlugins")
+sourceSets {
+    main {
+        java { srcDir("../paper-server/src/main/java") }
+        resources { srcDir("../paper-server/src/main/resources") }
+    }
+    test {
+        java { srcDir("../paper-server/src/test/java") }
+        resources { srcDir("../paper-server/src/test/resources") }
+    }
+}
+
+val log4jPlugins = sourceSets.create("log4jPlugins") {
+    java { srcDir("../paper-server/src/log4jPlugins/java") }
+}
+
 configurations.named(log4jPlugins.compileClasspathConfigurationName) {
     extendsFrom(configurations.compileClasspath.get())
 }
@@ -109,12 +_,12 @@
 }
 
 dependencies {
-    implementation(project(":paper-api"))
+    implementation(project(":kitty-api"))
     implementation("ca.spottedleaf:concurrentutil:0.0.10")
     implementation("org.jline:jline-terminal-ffm:3.27.1") // use ffm on java 22+
     implementation("org.jline:jline-terminal-jni:3.27.1") // fall back to jni on java 21
     implementation("net.minecrell:terminalconsoleappender:1.3.0")
-    implementation("net.kyori:adventure-text-serializer-ansi")
+    implementation("net.kyori:adventure-text-serializer-ansi:4.26.1")
 
     /*
       Required to add the missing Log4j2Plugins.dat file from log4j-core
@@ -157,6 +_,9 @@
     // Spark
     implementation("me.lucko:spark-api:0.1-20240720.200737-2")
     implementation("me.lucko:spark-paper:1.10.152")
+
+    // Kitty Paper
+    implementation("com.rabbitmq:amqp-client:5.27.1")
 }
 
 tasks.jar {
@@ -237,19 +_,12 @@
     jvmArgumentProviders.add(provider)
 }
 
-val generatedDir: java.nio.file.Path = layout.projectDirectory.dir("src/generated/java").asFile.toPath()
+val generatedDir: java.nio.file.Path = layout.projectDirectory.dir("../paper-server/src/generated/java").asFile.toPath()
 idea {
     module {
         generatedSourceDirs.add(generatedDir.toFile())
     }
 }
-sourceSets {
-    main {
-        java {
-            srcDir(generatedDir)
-        }
-    }
-}
 
 fun TaskContainer.registerRunTask(
     name: String,
@@ -315,19 +_,19 @@
     mainClass.set(null as String?)
 }
 
-fill {
-    project("paper")
-    versionFamily(paperweight.minecraftVersion.map { it.split(".", "-").takeWhile { part -> part.toIntOrNull() != null }.take(2).joinToString(".") })
-    version(paperweight.minecraftVersion)
-
-    build {
-        channel = providers.gradleProperty("channel").map { BuildChannel.valueOf(it.uppercase()) }
-
-        downloads {
-            register("server:default") {
-                file = tasks.createPaperclipJar.flatMap { it.outputZip }
-                nameResolver.set { project, _, version, build -> "$project-$version-$build.jar" }
-            }
-        }
-    }
-}
+//fill {
+//    project("paper")
+//    versionFamily(paperweight.minecraftVersion.map { it.split(".", "-").takeWhile { part -> part.toIntOrNull() != null }.take(2).joinToString(".") })
+//    version(paperweight.minecraftVersion)
+//
+//    build {
+//        channel = providers.gradleProperty("channel").map { BuildChannel.valueOf(it.uppercase()) }
+//
+//        downloads {
+//            register("server:default") {
+//                file = tasks.createPaperclipJar.flatMap { it.outputZip }
+//                nameResolver.set { project, _, version, build -> "$project-$version-$build.jar" }
+//            }
+//        }
+//    }
+//}
